### PR TITLE
feat: add support for reading config from file.

### PR DESCRIPTION
**Pull request by bloodearnest**
This centralised the configuration for osrelease tool, rather than
relying on per-user env vars. It means users need no special env vars,
and we can change config centrally as needed.

We drop support for the old env vars, as they are no longer needed.

### CODE REVIEW

File: publisher/config.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+def get_config(env=os.environ):
+    config = {}
+    config_file = get_config_file(env)
+    if config_file:
+        exec(config_file.read_text(), {}, config)
+    return config
+
+
+def get_private_github_token(env=os.environ):
```
Comment (sebbacon) [inline publisher/config.py:34]: Is this special method to give a better, newer name to private_repo_access_token?

If so, it's probably better to change it at source (in the example config) and optionally add a backwards compatibility shim in `get_config`?

Then we could have a more generic `get_config_value(variable_name)` here or similar?

**Review by sebbacon — APPROVED** [review]
I think this would also be the time to add some documentation about `osrelease_config.py`, its possible values and locations? In the README?

File: publisher/config.py (hunk 1 above)
Comment (bloodearnest) [reply publisher/config.py]: Good call, make these changes

**Comment by bloodearnest** [comment]
> I think this would also be the time to add some documentation about `osrelease_config.py`, its possible values and locations? In the README?

Good call, I have upated the README and added an example config

**Pull request merged**